CMakeLists.txt

```diff
@@ -124,7 +128,20 @@ list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 if(WIN32)
   add_library(ONEMKL::SYCL::SYCL INTERFACE IMPORTED)
 else()
-  find_package(Compiler REQUIRED)
+  # Find necessary packages
+  if (${ONEMKL_SYCL_IMPLEMENTATION} STREQUAL "hipSYCL")
```

Comment:
I think I'd prefer to use `hipsycl` to be in-line with using lowercase `dpc++`..
you could also use `string( TOLOWER "${ONEMKL_SYCL_IMPLEMENTATION}" ONEMKL_SYCL_IMPLEMENTATION)` beforehand, so the case of the input actually doesn't matter.
Also, if I understand it correctly, it is better to use `if(ONEMKL_SYCL_IMPLEMENTATION ..` without the dereferencing [source](https://stackoverflow.com/questions/33432432/cmake-why-makes-this-string-comparison-result-no-sense)

Reply:
I think I'm inclined to disagree about hipSYCL being lowercase. SYCL is a term that is inherently and strictly defined as all capital. DPC++ does not have such strong capitalization definitions. We also pretty consistently use the hipSYCL capitalization in all our stuff - the only exception being C++ source code inside hipSYCL.

In any case, actively turning everything lower case and hence accepting any capitalization seems like the best way to go.